eks/cluster-loader/addon.go

```diff
@@ -0,0 +1,101 @@
+package clusterloader
+
+import (
+	"fmt"
+	"github.com/aws/aws-k8s-tester/eksconfig"
+	k8sclient "github.com/aws/aws-k8s-tester/pkg/k8s-client"
+	gotemplate "github.com/aws/aws-k8s-tester/pkg/util"
+	"io/ioutil"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"net/http"
+	"strings"
+)
+
+type ClusterLoader struct {
+	Config    *eksconfig.Config
+	K8sClient k8sclient.EKS
+}
+
+type ConfigMapFileStore struct {
+	Metadata v1.ObjectMeta
+	Data     map[string]string
+}
+
+// IsEnabled returns true if enabled
+func (c *ClusterLoader) IsEnabled() bool {
+	return c.Config.Spec.ClusterLoader != nil
+}
+
+// Apply installs the addon
+func (c *ClusterLoader) Apply() (err error) {
+	if c.Config.Spec.ClusterLoader.TestParams != nil {
+		for i, elem := range c.Config.Spec.ClusterLoader.TestParams {
+			if i < len(c.Config.Spec.ClusterLoader.TestParams)-1 {
+				c.Config.Spec.ClusterLoader.TestParams[i] = elem + ","
+			}
+		}
+	}
+	configmap := ConfigMapFileStore{
+		Metadata: v1.ObjectMeta{
+			Name: "cl2-config",
+		},
+		Data: c.formDataMap(),
+	}
+
+	template, err := gotemplate.FromLocalDirectory(struct {
+		*eksconfig.ClusterLoaderSpec
+		ConfigMap ConfigMapFileStore
+	}{
+		ClusterLoaderSpec: c.Config.Spec.ClusterLoader,
+		ConfigMap:         configmap,
+	})
+	c.K8sClient.Delete(template.String())
+	if err != nil {
+		return fmt.Errorf("while building templates, %v", err)
+	}
+	if err := c.K8sClient.Apply(template.String()); err != nil {
+		return fmt.Errorf("while applying resources, %v", err)
+	}
+	c.Config.Status.ClusterLoader = &eksconfig.ClusterLoaderStatus{
+		AddonStatus: eksconfig.AddonStatus{
+			Installed: true,
+			Ready:     true,
+		},
+	}
+	return nil
+}
+
+// Delete removes the addon
+func (c *ClusterLoader) Delete() (err error) {
+	template, err := gotemplate.FromLocalDirectory(c.Config.Spec.ClusterLoader)
+	if err != nil {
+		return fmt.Errorf("while building templates, %v", err)
+	}
+	if err := c.K8sClient.Delete(template.String()); err != nil {
+		return fmt.Errorf("while deleting resources, %v", err)
+	}
+	c.Config.Status.MetricsServer = &eksconfig.MetricsServerStatus{
+		AddonStatus: eksconfig.AddonStatus{
+			Installed: false,
+			Ready:     false,
+		},
+	}
+	return nil
+}
+
+func (c *ClusterLoader) formDataMap() map[string]string {
```

Comment:
It's pretty common to name functions like you've done `<verb><noun>, <form><datamap>`. However, "form" isn't typically used. Much more common would be "new", "build" or "create" depending on the situation. Create often implies that something significant is being created, such as a large object is constructed or an http call is being made. 

I'd also avoid saying "datamap". Your function returns a map, so having map in the name stutters. Following your line of logic, you might start calling this function functionDataMap(). I'd probably call this buildConfigMapData(). In this case, the "map" configmap refers explicitly to the kubernetes object and not the map data structure. If you wanted to be more brief, you could just call it "buildData" and let the reader infer the context.